Implement netbox_tenant module

amylieb opened the pull request:
##### SUMMARY
I've started leveraging @FragmentedPacket's netbox modules and found I wanted a "netbox_tenant" module, so I copied netbox_site and altered it to manipulate the tenant endpoint.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
netbox_tenant - module for getting/creating 'tenant' netbox endpoints.

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
Adapting netbox_site to create netbox_tenant was straightforward. No changes to [netbox_utils](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/net_tools/netbox/netbox_utils.py) were needed.
<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

amylieb committed:
Implement netbox_tenant module

jillr commented:
Thanks for this contribution!
It looks like there's some test failures that will need to be addressed for this module.  You can run the command the bot is showing above each of the test failures on your local machine to test if changes you make will pass the tests before pushing any fixes up to github.

We'd also appreciate if you would be willing to write integration tests for the new module.  Integration tests generally take the form of ansible playbooks that exercise the functionality of the module, accounting for different options or configurations as necessary.  They go in a specific directory structure that allows CI to know about them and execute them properly.  Unfortunately it looks like we don't yet have any tests for the other netbox modules to compare to, but we have some general info here:
https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html#testing-integration
And you might be able to get some useful info from some of the other network tools that have integration tests, like https://github.com/ansible/ansible/tree/devel/test/integration/targets/nios_network  or https://github.com/ansible/ansible/tree/devel/test/integration/targets/slurp.

FragmentedPacket commented:
@jillr I'm currently working on refactoring the `netbox_utils` and then increasing the amount of tests and types of tests.

I'd like to do integration testing as that will provide better coverage. Is there anyway to have the tests spin up some docker containers for Netbox to run the tests against?

amylieb commented:
Fixed sanity check issues. Am more than happy to write integration tests, but any meaningful test would require a Netbox instance to test against. Echoing @FragmentedPacket's request to be able to spin up Netbox instance(s) in containers as part of the testing process.

jillr commented:
That's great to hear @FragmentedPacket, we can do docker images but it's a bit more work to get setup so we'd want to do that in a separate PR.  You can take a look at the cloudstack (cs_ prefix) test suite as an example.  So for this PR no tests would be totally fine if they're in the works.

FragmentedPacket commented:
@amylieb Hey Amy. I refactored the modules and moved them to Ansible Collections due to that being their direction so not planning to maintain/add anything new into core.

Here is the link to the Ansible Collection: https://galaxy.ansible.com/fragmentedpacket/netbox_modules

I did incorporate tenant and tenant_group with you as the author. Thanks for taking the time to create that and hope to see you contribute further!

amylieb commented:
Thanks Mikhail!

While the project that generated my specific use case for these modules has
been terminated, there is a very good chance we'll be doing more with
netbox ansible modules in the future.

- Amy

On Fri, Sep 13, 2019 at 4:56 PM Mikhail Yohman <notifications@github.com>
wrote:

> @amylieb <https://github.com/amylieb> Hey Amy. I refactored the modules
> and moved them to Ansible Collections due to that being their direction so
> not planning to maintain/add anything new into core.
>
> Here is the link to the Ansible Collection:
> https://galaxy.ansible.com/fragmentedpacket/netbox_modules
>
> I did incorporate tenant and tenant_group with you as the author. Thanks
> for taking the time to create that and hope to see you contribute further!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ansible/ansible/pull/58140?email_source=notifications&email_token=AEVHKKPZJMN6K2YJHLQSRFTQJP5BFA5CNFSM4HZ6ADDKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6WFLBY#issuecomment-531387783>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEVHKKMASVTYH2XJSHVQM3TQJP5BFANCNFSM4HZ6ADDA>
> .
>


-- 
Amy Liebowitz
402-517-0220
Network Architect Senior
University of Michigan

Akasurde commented:
@amylieb Thanks for the contribution. This is implemented via https://github.com/netbox-community/ansible_modules/pull/7.

Akasurde commented:
Hi @amylieb, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/netbox-community/ansible_modules<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>
